### PR TITLE
Move IPCs from being shown on medhud to diagnostic hud

### PR DESCRIPTION
**Pull request by moxian**
## What Does This PR Do
IPC healthbar is no longer shown when you have medhud on
IPC healthbar is now shown to you when you have diagnostic hud on

## Why It's Good For The Game
This was a requested change in discord.
I'll let the people in favour of this explain the reasoning in comments.

## Images of changes
diagnostic hud:
20220529-221514-dreamseeker
medical hud:
20220529-221521-dreamseeker

## Changelog
:cl:
tweak: IPC health bar is now shown on diagnostic hud rather than medical
/:cl:

### CODE REVIEW

File: code/game/data_huds.dm [hunk 1]
```diff
@@ -181,12 +184,16 @@
 //called when a carbon changes stat, virus or XENO_HOST
 /mob/living/carbon/med_hud_set_status()
 	var/image/holder = hud_list[STATUS_HUD]
+	if(istype(dna?.species, /datum/species/machine))
```
Comment (Sirryan2002) [inline code/game/data_huds.dm:187]:
```suggestion
	if(ismachineperson(dna?.species))
```
helper already exists for this
`#define ismachineperson(A) (is_species(A, /datum/species/machine))`

File: code/game/data_huds.dm [hunk 2]
```diff
@@ -181,12 +184,16 @@
...
+		holder = hud_list[DIAG_STAT_HUD]
 	var/mob/living/simple_animal/borer/B = has_brain_worms()
 	// To the right of health bar
 	if(stat == DEAD || HAS_TRAIT(src, TRAIT_FAKEDEATH))
 		var/revivable
 		if(!ghost_can_reenter()) // DNR or AntagHUD
 			revivable = FALSE
+		else if(istype(dna.species, /datum/species/machine))
```
Comment (Sirryan2002) [inline code/game/data_huds.dm:195]:
```suggestion
		else if(ismachineperson(dna.species))
```

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
PR currently fails CI
```
[2022-05-29T22:24:48] Runtime in data_huds.dm,[174](https://github.com/ParadiseSS13/Paradise/runs/6646239290?check_suite_focus=true#step:9:175): Cannot modify null.icon_state.
  proc name: med hud set health (/mob/living/proc/med_hud_set_health)
  src: PBU-318 (/mob/living/carbon/brain)
  src.loc: the robotic brain (/obj/item/mmi/robotic_brain)
  call stack:
  PBU-318 (/mob/living/carbon/brain): med hud set health()
  PBU-318 (/mob/living/carbon/brain): prepare data huds()
  PBU-318 (/mob/living/carbon/brain): prepare huds()
  PBU-318 (/mob/living/carbon/brain): Initialize(1)
  PBU-318 (/mob/living/carbon/brain): Initialize(1)
  PBU-318 (/mob/living/carbon/brain): Initialize(1)
  Atoms (/datum/controller/subsystem/atoms): InitAtom(PBU-318 (/mob/living/carbon/brain), /list (/list))
  Atoms (/datum/controller/subsystem/atoms): InitializeAtoms(null, 1)
  Atoms (/datum/controller/subsystem/atoms): Initialize(806881)
  Master (/datum/controller/master): Initialize(10, 0, 1)
[2022-05-29T22:24:48] Runtime in data_huds.dm,214: Cannot modify null.icon_state.
  proc name: med hud set status (/mob/living/carbon/med_hud_set_status)
  src: PBU-318 (/mob/living/carbon/brain)
  src.loc: the robotic brain (/obj/item/mmi/robotic_brain)
  call stack:
  PBU-318 (/mob/living/carbon/brain): med hud set status()
  PBU-318 (/mob/living/carbon/brain): prepare data huds()
  PBU-318 (/mob/living/carbon/brain): prepare huds()
  PBU-318 (/mob/living/carbon/brain): Initialize(1)
  PBU-318 (/mob/living/carbon/brain): Initialize(1)
  PBU-318 (/mob/living/carbon/brain): Initialize(1)
  Atoms (/datum/controller/subsystem/atoms): InitAtom(PBU-318 (/mob/living/carbon/brain), /list (/list))
  Atoms (/datum/controller/subsystem/atoms): InitializeAtoms(null, 1)
  Atoms (/datum/controller/subsystem/atoms): Initialize(806881)
  Master (/datum/controller/master): Initialize(10, 0, 1)
```

File: code/game/data_huds.dm [hunk 3]
```diff
@@ -86,6 +86,8 @@
 		return TRUE
 	if(undergoing_cardiac_arrest())
 		return TRUE
+	if(istype(dna?.species, /datum/species/machine) && health < 0)
```
Comment (Sirryan2002) [inline code/game/data_huds.dm:89]:
```suggestion
	if(ismachineperson(dna?.species) && health < 0)
```

File: code/game/data_huds.dm [hunk 4]
```diff
@@ -167,9 +169,10 @@
 //called when a living mob changes health
 /mob/living/proc/med_hud_set_health()
 	var/image/holder = hud_list[HEALTH_HUD]
+	if(istype(dna?.species, /datum/species/machine))
```
Comment (Sirryan2002) [inline code/game/data_huds.dm:172]:
```suggestion
	if(ismachineperson(dna?.species))
```

**Comment by moxian** [comment]
fails no more

**Comment by A-Medster** [comment]
I requested this, this helps the game because it gives the ability to see an IPC's health to the people actually fixing them. A roboticist can see how close they are to a fully repaired IPC or just see that an IPC is damaged and need repairs. Taking them off of the med hud wasnt totally my idea but make sense given they cant repair IPCs any way.

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**